<launch>
  <!-- Reuse existing arguments -->
  <arg name="urdf_path" default="/home/palguno/catkin_ws/src/manip2_urdf_description/urdf/manip2_urdf.xacro"/>
  <arg name="port_name" default="/dev/ttyUSB0"/>
  <arg name="baud" default="1000000"/>
  <arg name="read_rate" default="6"/>
  <arg name="print_reads" default="true"/>
  <arg name="publish_error_text" default="true"/>
  <arg name="p2_hw_error_addr" default="892"/>

  <!-- New: telemetry loop rate -->
  <arg name="telem_rate" default="5.0"/>

  <!-- Robot description -->
  <param name="robot_description" command="$(find xacro)/xacro $(arg urdf_path)"/>

  <!-- Robot state publisher -->
  <!--  <node pkg="robot_state_publisher" type="robot_state_publisher" name="robot_state_publisher" output="screen">
    <param name="publish_frequency" value="30.0"/>
  </node> -->

  <!-- Telemetry-enabled bridge -->
  <!-- If you merged telemetry into manip2_dxl_bridge.py, keep 'type' as that file instead. -->
  <node pkg="manip2_control" type="manip2_dxl_bridge_telem" name="manip2_dxl_telem" output="screen">
    <param name="port_name" value="$(arg port_name)"/>
    <param name="baud_rate" value="$(arg baud)"/>

    <param name="dxl_ids" value="[1,2,3,4,5]"/>
    <param name="dxl_id_p2" value="2"/>

    <param name="speed_p1_value" value="60"/>
    <param name="speed_p2_value" value="450"/>

    <param name="urdf_limits_deg" value="[[-90,90],[-30,50],[-105,0],[-77,113],[-103,103]]"/>
    <param name="tick_offsets_nom" value="[486,-3030,1722,509,506]"/>
    <param name="degs_per_tick_nom" value="[0.29296875,0.001185185,0.0612916,-0.29296875,0.29296875]"/>

    <param name="read_rate_hz" value="$(arg read_rate)"/>
    <param name="print_reads" value="$(arg print_reads)"/>
    
    <param name="p2_hw_error_addr" value="$(arg p2_hw_error_addr)"/>
    <param name="publish_error_text" value="$(arg publish_error_text)"/>

    <!-- New: telemetry loop rate param consumed by the telemetry thread -->
    <param name="telemetry_rate_hz" value="$(arg telem_rate)"/>

    <param name="home_deg" value="[0,0,0,0,0]"/>
  </node>
</launch>
